### PR TITLE
fix(next): next_entity_type_config dependencies

### LINKED ISSUE #599: NextEntityTypeConfig does not calculate dependencies
### Package containing the bug

next (Drupal module)

### Expected behavior

Deleting a content type should delete the matching NextEntityTypeConfig config entity. It does not because there are no calculated dependencies.

```yaml
dependencies: {  }
id: node.page
```

### Steps to reproduce:

1. First create content type
1. Then create NextEntityTypeConfig
1. Delete content type
1. 😢 expect NextEntityTypeConfig to be removed

### PR DESCRIPTION
**Pull request by mglaman**
Fixes #599,  dependencies in next_entity_type_config for the entity type and bundle being targeted.

This pull request is for: (mark with an "x")

- [ ] `examples/*`
- [x] `modules/next`
- [ ] `packages/next-drupal`
- [ ] `starters/basic-starter`
- [ ] `starters/graphql-starter`
- [ ] Other

GitHub Issue: #599
_Please add a link to the GitHub issue
where this problem is discussed._

- [ ] I need help adding tests. (mark with an "x")
      _Code changes need test coverage. If you don't know
      how to make tests, check this box to ask for help._

## Describe your changes

A clear and concise description of what the request is.

If applicable, add screenshots to help explain your issue.

### CODE REVIEW

File: modules/next/tests/src/Kernel/Entity/NextEntityTypeConfigTest.php [hunk 1]
```diff
@@ -140,4 +140,27 @@ public function testRevalidator() {
...
+    $entity_type_config = NextEntityTypeConfig::create([
+      'id' => 'node.page',
+      'site_resolver' => 'site_selector',
+      'configuration' => [
+        'sites' => [
+          'blog' => 'blog',
+        ],
+      ],
+    ]);
+    self::assertEquals([], $entity_type_config->getDependencies());
```
Comment (JohnAlbin) [inline modules/next/tests/src/Kernel/Entity/NextEntityTypeConfigTest.php:163]: Am I misreading this assertion?

If the purpose of `calculateDependencies()` is to add the proper dependencies to the config, why is the test asserting that `getDependencies()` is an empty array? How does this test that dependencies are added properly?

Comment (mglaman) [reply modules/next/tests/src/Kernel/Entity/NextEntityTypeConfigTest.php:163]: Good question. I can't remember if I left this here to show the test was still failing or what... Or if I wrote the test first and then the fix?

I'll follow back up on this today

**Comment by mglaman** [comment]
@JohnAlbin I tried running locally, but for some reason all tests fail now. So I wanted to use the CI... but it looks like `next.yml` workflow may need approval.

**Comment by mglaman** [comment]
Okay, it should be all set, now
